### PR TITLE
Feature/switch tweet form bot conversation

### PR DESCRIPTION
**Pull request by ChristopherKchilton**
# Pull Request Template

## Description

This changed the send form function which is tied to the form out api, to check the form variable. If the form variable is 0 the twitter bot will either send a link inviting people on twitter to fill out a form to provide more information regarding the incident. This is based of the initial incident tweet ID. If the form variable is 1 it will send a direct message request in a tweet to the initial tweeter. From there the twitter bot can start having a conversation. (This is the kick off point for a more conversational option for the twitter bot)

https://www.loom.com/share/6e9cd08a74714b9b91fde52c897f7df7

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] I tested the code using the form out API on the DS side to test the functionality of the code. If the form has a 0 it sends a tweet asking for them to fill out a form via a link. If the form variable has a 1 it send a direct message request link which currently starts the bot 3 step conversation.
- [ ] Further testing may need to be done to test if multiple requests could be sent if the admin requested them. 

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: app/bot.py [hunk 1]
```diff
@@ -38,23 +38,36 @@
...
+	form_link = f'https://a.humanrightsfirst.dev/edit/{incident_id}'
 	to_insert['tweet_id'] = int(to_insert['tweet_id'])
-	to_insert['tweet_text'] = '@' + to_insert['tweeter_id'] + ' ' + conversation_tree[10] + '\n' + dm_link 
-	to_insert['reachout_template'] = conversation_tree[10]
 	to_insert['in_reply_to_id'] = to_insert['tweeter_id']
 	del to_insert['link']
-
-
-	status = twitter.respond_to_tweet(to_insert['tweet_id'], to_insert['tweet_text'])
-	to_insert['tweeter_id'] = bot_name
```
Comment (iknight7000) [inline app/bot.py]: I can see where we are removing redundant code here to match the the changes you've made regarding the split.

**Review by iknight7000 — APPROVED** [review]
The update to the send form will allow for more scalability in the future, which helps solve our current dilemma with everything being poorly designed.

File: app/twitter.py [hunk 2]
```diff
@@ -112,18 +105,10 @@ def get_replies(user_id: str, tweet_id: int, replier_id: str) -> str:
...
-        except tweepy.TweepError as e:
-            logging.error("Tweepy error occured:{}".format(e))
-            break
         except StopIteration:
             print('StopIteration')
             break
-        except Exception as e:
-            logger.error("Failed while fetching replies {}".format(e))
-            break
+
```
Comment (gaben3722) [inline app/twitter.py:112]: The amount of code you were able to condense is amazing great job!

**Review by iknight7000 — APPROVED** [review]

**Review by iknight7000 — APPROVED** [review]
Removing the try statements makes this code easier to refactor later, and changes the conversation. These statements seemed more like temporary fixes, not really scalable.

**Review by iknight7000 — APPROVED** [review]

**Review by paulstgermain — APPROVED** [review]
Approving based on meetings with the developer, confirming these changes are working. Dropping more feedback at a later date as timely merging is important in this case.

**Pull request merged**